test/integration/framework/master_utils.go

```diff
@@ -165,6 +165,8 @@ func startMasterOrDie(masterConfig *master.Config, incomingServer *httptest.Serv
 		Groups: []string{user.SystemPrivilegedGroup},
 	}
 
+	masterConfig.GenericConfig.PublicPort = 443
```

Comment:
now I'm confused about what this is used for... is this the external port (defaulting to 6443) or the service port (defaulting to 443)? defaulting to the --secure-port (e.g. the bind port) would mean this should be 6443, right?

Reply:
It's the external port. APIServerServicePort is the port of the service. The defaulting happens in ServerRunOptions which integration test do not use when constructing the master config:

https://github.com/kubernetes/kubernetes/blob/de8fb1c9f81b1bd48a0d25f866492acca2c199f0/test/integration/framework/master_utils.go#L271

The integration test that is failing:

https://github.com/kubernetes/kubernetes/blob/de8fb1c9f81b1bd48a0d25f866492acca2c199f0/test/integration/master/synthetic_master_test.go#L640

With a validation error like "port of endpoint can't be 0".

It doesn't matter too much what it is since no other integration tests use it but it may cause log spam if not set. I can either:
* Leave it here somewhere around here.
* Try to use ServerRunOptions here.
* Default it somewhere deeper than Options.

Reply:
Can we not use the word "public" and use either "bind", "advertise", or "service"?